system_tests/tests/tjmper.py

```diff
@@ -0,0 +1,145 @@
+import contextlib
+import unittest
+
+from parameterized import parameterized
+
+from utils.channel_access import ChannelAccess
+from utils.ioc_launcher import get_default_ioc_dir
+from utils.test_modes import TestModes
+from utils.testing import get_running_lewis_and_ioc, parameterized_list
+
+
+DEVICE_PREFIX = "TJMPER_01"
+
+
+IOCS = [
+    {
+        "name": DEVICE_PREFIX,
+        "directory": get_default_ioc_dir("TJMPER"),
+        "macros": {},
+        "emulator": "Tjmper",
+    },
+]
+
+
+TEST_MODES = [TestModes.RECSIM, TestModes.DEVSIM]
+
+
+MODES = [
+    ("All out",                 [1, 0, 1, 0, 1, 0]),
+    ("PLT1 and SMPL engaged",   [0, 1, 1, 0, 0, 1]),
+    ("PLT2 and SMPL engaged",   [1, 0, 0, 1, 0, 1])
+]
+
+MOVING_PVS = [
+    "PLATE1:MOVING",
+    "PLATE2:MOVING",
+    "SAMPLE:MOVING"
+]
+
+AIR_SUPPLY = [
+    (0, "Off Dumped"),
+    (1, "Off Valve Reset"),
+    (2, "On Dumped"),
+    (3, "On Valve Reset")
+]
+
+ERRORS = [
+    (0, "No Error (Operational)",   ChannelAccess.Alarms.NONE),
+    (1, "PLT1 Homing TMO",          ChannelAccess.Alarms.MAJOR),
+    (2, "PLT1 Engaging TMO",        ChannelAccess.Alarms.MAJOR),
+    (3, "PLT1EN SMPL Homing TMO",   ChannelAccess.Alarms.MAJOR),
+    (4, "PLT1EN SMPL Engaging TMO", ChannelAccess.Alarms.MAJOR),
+    (5, "PLT2 Homing TMO",          ChannelAccess.Alarms.MAJOR),
+    (6, "PLT2 Engaging TMO",        ChannelAccess.Alarms.MAJOR),
+    (7, "PLT2EN SMPL Homing TMO",   ChannelAccess.Alarms.MAJOR),
+    (8, "PLT2EN SMPL Engaging TMO", ChannelAccess.Alarms.MAJOR)
+]
+
+ALARM_PVS = [
+    "ID",
+    "MODE",
+    "LMT",
+    "LMT:PLATE1:HOME",
+    "LMT:PLATE1:ENGAGED",
+    "LMT:PLATE2:HOME",
+    "LMT:PLATE2:ENGAGED",
+    "LMT:SAMPLE:HOME",
+    "LMT:SAMPLE:ENGAGED",
+    "AIR",
+    "ERR"
+]
+
+
+class TjmperTests(unittest.TestCase):
+    """
+    Tests for the Tjmper IOC.
+    """
+    def setUp(self):
+        self._lewis, self._ioc = get_running_lewis_and_ioc("Tjmper", DEVICE_PREFIX)
+        self._lewis.backdoor_run_function_on_device("reset")
+        self.ca = ChannelAccess(device_prefix=DEVICE_PREFIX, default_wait_time=0)
+
+    def test_WHEN_id_set_via_backdoor_THEN_id_updates(self):
+        id_value = 4
+        self._lewis.backdoor_set_on_device("id", id_value)
+        self.ca.assert_that_pv_is("ID", id_value)
+
+    @parameterized.expand(parameterized_list(MOVING_PVS))
+    def test_WHEN_limits_set_via_backdoor_THEN_moving_pvs_update(self, _, moving_pv):
+        self._lewis.backdoor_set_on_device("plate_1_home", 0)
+        self._lewis.backdoor_set_on_device("plate_1_engaged", 0)
+        self._lewis.backdoor_set_on_device("plate_2_home", 0)
+        self._lewis.backdoor_set_on_device("plate_2_engaged", 0)
+        self._lewis.backdoor_set_on_device("sample_home", 0)
+        self._lewis.backdoor_set_on_device("sample_engaged", 0)
```

Comment:
things like this won't work in RECSIM as RECSIM doesn't use lewis, so best to either decorate with @skip_if_recsim or alternatively just remove RECSIM from the TEST_MODES list at the top of the file